src/main/java/com/treasuredata/client/TDClientConfig.java

```diff
@@ -0,0 +1,488 @@
+            this.endpoint = getConfigProperty(TD_CLIENT_API_ENDPOINT, defaultValues);
+            this.port = getConfigPropertyInt(TD_CLIENT_API_PORT, defaultValues);
+            this.useSSL = Boolean.parseBoolean(getConfigProperty(TD_CLIENT_USESSL, defaultValues).or("false"));
+
+            // For APIKEY we read the environment variable.
+            // We also read apikey, user and password specified in td.conf file
+            this.apiKey = Optional.fromNullable(System.getenv(ENV_TD_CLIENT_APIKEY))
+                    .or(getConfigProperty(TD_CLIENT_APIKEY, defaultValues))
+                    .or(Optional.fromNullable(defaultValues.getProperty("apikey")));
+            this.user = getConfigProperty(TD_CLIENT_USER, defaultValues)
+                    .or(Optional.fromNullable(defaultValues.getProperty("user")));
+            this.password = getConfigProperty(TD_CLIENT_PASSOWRD, defaultValues)
+                    .or(Optional.fromNullable(defaultValues.getProperty("password")));
+
+            // proxy
+            Optional<String> proxyHost = getConfigProperty(TD_CLIENT_PROXY_HOST, defaultValues);
+            Optional<Integer> proxyPort = getConfigPropertyInt(TD_CLIENT_PROXY_PORT, defaultValues);
+            Optional<String> proxyUser = getConfigProperty(TD_CLIENT_PROXY_USER, defaultValues);
+            Optional<String> proxyPassword = getConfigProperty(TD_CLIENT_PROXY_PASSWORD, defaultValues);
+            boolean hasProxy = false;
+            ProxyConfig.ProxyConfigBuilder proxyConfig = new ProxyConfig.ProxyConfigBuilder();
+            if (proxyHost.isPresent()) {
+                hasProxy = true;
+                proxyConfig.setHost(proxyHost.get());
+            }
+            if (proxyPort.isPresent()) {
+                hasProxy = true;
+                proxyConfig.setPort(proxyPort.get());
+            }
+            if (proxyUser.isPresent()) {
+                hasProxy = true;
+                proxyConfig.setUser(proxyUser.get());
+            }
+            if (proxyPassword.isPresent()) {
+                hasProxy = true;
+                proxyConfig.setPassword(proxyPassword.get());
+            }
+            this.proxy = Optional.fromNullable(hasProxy ? proxyConfig.createProxyConfig() : null);
+
+            // http client parameter
+            this.retryLimit = getConfigPropertyInt(TD_CLIENT_RETRY_LIMIT, defaultValues).or(7);
+            this.retryInitialWaitMillis = getConfigPropertyInt(TD_CLIENT_RETRY_INITIAL_WAIT_MILLIS, defaultValues).or(1000);
+            this.retryIntervalMillis = getConfigPropertyInt(TD_CLIENT_RETRY_INTERVAL_MILLIS, defaultValues).or(2000);
+            this.connectTimeoutMillis = getConfigPropertyInt(TD_CLIENT_CONNECT_TIMEOUT_MILLIS, defaultValues).or(15000);
+            this.idleTimeoutMillis = getConfigPropertyInt(TD_CLIENT_IDLE_TIMEOUT_MILLIS, defaultValues).or(60000);
+            this.connectionPoolSize = getConfigPropertyInt(TD_CLIENT_CONNECTION_POOL_SIZE, defaultValues).or(64);
```

Comment:
64 is the default value of Jetty and it does not open 64 threads at once. So if client uses only a single connection, it will create just 1 thread. 

If you still think this value should be a small value (e.g., 1, 4), let me know.